data-model/src/main/java/org/pdxfinder/graph/repositories/ModelCreationRepository.java

```diff
@@ -208,19 +208,21 @@ Integer countMarkerAssociationBySourcePdxId(@Param("sourcePdxId") String sourceP
     List<ModelCreation> findModelPlatformSampleByDS(@Param("dataSource") String dataSource);
 
 
-    @Query("MATCH (mod:ModelCreation) WHERE toLower(mod.dataSource) = toLower({dataSource}) and mod.sourcePdxId = {modelId} " +
+    @Query("MATCH (mod:ModelCreation) WHERE toLower(mod.dataSource) = toLower({dataSource}) " +
             "WITH mod " +
-            "OPTIONAL MATCH (mod)-[iir:IMPLANTED_IN]-(psamp:Sample)-[cbr:CHARACTERIZED_BY]-(mc:MolecularCharacterization)-[assoc:ASSOCIATED_WITH]->(mAss:MarkerAssociation) " +
+            "MATCH (mod)--(samp:Sample)-[cbr:CHARACTERIZED_BY]-(mc:MolecularCharacterization)-[assoc:ASSOCIATED_WITH]->(mAss:MarkerAssociation) " +
```

Comment:
If this is not an optional match, then only those samples will be returned here that have molecularcharacterization with data. Sometimes we have molchar nodes without markerassociation (so we can say that the model has molecular data just not in the finder). Disregard this if this is the intended behaviour.

Reply:
Also, (mod)--(samp:Sample) will return patient and xenograft samples as well.

Reply:
I want to grab all samples with marker associations so the data can be written into an omic data sheet. I want all the models with marker associations on either the patient sample or the models sample. However, I also want all the information necessary for the object mapper to fill out all the necessary data. I'm not sure what is necessary and what is not for the ModelCreation to get filled completely, but it seems to work.